engine/src/main/java/org/terasology/recording/RecordedEventSerializer.java

```diff
@@ -0,0 +1,515 @@
+            writer.beginObject();
+            array = data.getElement().getAsJsonArray();
+            writer.name("x").value(array.get(0).getAsFloat());
+            writer.name("y").value(array.get(1).getAsFloat());
+            writer.name("z").value(array.get(2).getAsFloat());
+            writer.endObject();
+
+            handler = typeSerializationLibrary.getTypeHandlerFromClass(Vector3i.class);
+            data = (GsonPersistedData) handler.serialize(e.getTargetBlockPosition(), serializationContext);
+            writer.name("targetBlockPosition");
+            writer.beginObject();
+            array = data.getElement().getAsJsonArray();
+            writer.name("x").value(array.get(0).getAsInt());
+            writer.name("y").value(array.get(1).getAsInt());
+            writer.name("z").value(array.get(2).getAsInt());
+            writer.endObject();
+
+            writeInputEventInstanceData(writer, event, serializationContext);
+
+
+        } else if (event instanceof CameraTargetChangedEvent) {
+            CameraTargetChangedEvent e = (CameraTargetChangedEvent) event;
+            writer.name("OldTarget").value(e.getOldTarget().getId());
+            writer.name("NewTarget").value(e.getNewTarget().getId());
+        } else if (event instanceof PlaySoundEvent) {
+            PlaySoundEvent e = (PlaySoundEvent) event;
+            writer.name("volume").value(e.getVolume());
+            TypeHandler handler = typeSerializationLibrary.getTypeHandlerFromClass(StaticSound.class);
+            PersistedData data = handler.serialize(e.getSound(), serializationContext);
+            writer.name("sound").value(data.getAsString());
+
+        } else if (event instanceof CharacterMoveInputEvent) {
+            CharacterMoveInputEvent e = (CharacterMoveInputEvent) event;
+            writer.name("delta").value(e.getDeltaMs());
+            writer.name("pitch").value(e.getPitch());
+            writer.name("yaw").value(e.getYaw());
+            writer.name("running").value(e.isRunning());
+            writer.name("crouching").value(e.isCrouching());
+            writer.name("jumpRequested").value(e.isJumpRequested());
+            writer.name("sequeceNumber").value(e.getSequenceNumber());
+            writer.name("firstRun").value(e.isFirstRun());
+            TypeHandler handler = typeSerializationLibrary.getTypeHandlerFromClass(Vector3f.class);
+            GsonPersistedData data = (GsonPersistedData) handler.serialize(e.getMovementDirection(), serializationContext);
+            writer.name("movementDirection");
+            writer.beginObject();
+            JsonArray array = data.getElement().getAsJsonArray();
+            writer.name("x").value(array.get(0).getAsFloat());
+            writer.name("y").value(array.get(1).getAsFloat());
+            writer.name("z").value(array.get(2).getAsFloat());
+            writer.endObject();
+
+        } else {
+            logger.error("ERROR: EVENT NOT SUPPORTED FOR SERIALIZATION");
+        }
```

Comment:
strategy pattern or something similar instead of if-else cascade ;) same concept as above